SingularityService/src/main/java/com/hubspot/singularity/mesos/SingularityMesosStatusUpdateHandler.java

```diff
@@ -344,6 +346,35 @@ private StatusUpdateResult unsafeProcessStatusUpdate(
           taskIdObj
         );
         return StatusUpdateResult.KILL_TASK;
+      } else if (
+        maybeTaskHistory.isPresent() &&
+        status.getReason() == Reason.REASON_AGENT_REREGISTERED
+      ) {
+        boolean lbRemovalStarted = maybeTaskHistory
+          .get()
+          .getLoadBalancerUpdates()
+          .stream()
+          .anyMatch(
+            lbUpdate ->
+              lbUpdate.getMethod() == LoadBalancerMethod.CANCEL ||
+              lbUpdate.getMethod() == LoadBalancerMethod.DELETE
```

Comment:
These are the methods we cal in baragon, not a representation of an add or remove. What we want to be checking is for a `LoadBalancerRequestType` of `REMOVE`. taskManager has a `getLoadBalancerState` method for this